torch/optim/adagrad.py

```diff
@@ -12,24 +12,28 @@ class Adagrad(Optimizer):
         params (iterable): iterable of parameters to optimize or dicts defining
             parameter groups
         lr (float, optional): learning rate (default: 1e-2)
+        eps (float, optional): term added to the denominator to improve
+            numerical stability (default: 1e-10)
         lr_decay (float, optional): learning rate decay (default: 0)
         weight_decay (float, optional): weight decay (L2 penalty) (default: 0)
 
     .. _Adaptive Subgradient Methods for Online Learning and Stochastic
         Optimization: http://jmlr.org/papers/v12/duchi11a.html
     """
 
-    def __init__(self, params, lr=1e-2, lr_decay=0, weight_decay=0, initial_accumulator_value=0):
+    def __init__(self, params, lr=1e-2, lr_decay=0, eps=1e-10, weight_decay=0, initial_accumulator_value=0):
```

Comment:
better to put new arg to the end (for backward compatibility)

Reply:
I ordered them this way initially to match the other optimizers, but yeah putting it at the end is better for bc.